### DISCUSSION TITLE
Upgrade from version 2 to 3 breaks pagination

### DISCUSSION THREAD

**Discussion by MomoDerDodo** [livewire/livewire#7143] [7. Bugs (archived)] [answered]
### Livewire version

3

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

Setup a Project with Livewire v2.12 and create a livewire component with pagination. Then upgrade livewire to v3 and the pagination will be broken.

I added 2 Screenshots to show the Pagination only changing the first element, when using the controls.

![Before Click](https://github.com/livewire/livewire/assets/38343956/e3f755cc-613a-4f40-92f8-3e7ae34e43e6)
![After Click](https://github.com/livewire/livewire/assets/38343956/47e86a35-a96d-44b3-8241-b007cf790011)

### Additional information

If i create a new Project and install version 3 directly the pagination works fine. If i upgrade from 2 to 3 it breaks all the time. I tried it with a fresh project and 2 older ones.

Controlling the pagination directly through the URL does work but using the generated controls changes just the first element that is shown and the others stay the same.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by DanielSpravtsev** [comment]
Publish new pagination view vendor files

**Reply by MomoDerDodo** [reply]
I did that, that was not the problem. Without doing that there would be an error.

**Reply by DanielSpravtsev** [reply]
> I did that, that was not the problem. Without doing that there would be an error.

Then it looks like you trying to render pagination inside foreach loop, when it should be outside of it.

**Reply by MomoDerDodo** [reply]
I'm not doing that.

This is the Component of my test case:
```blade
<div>
    <div class="grid grid-cols-1 gap-5">
        @forelse($posts as $post)
            <div class="border rounded p-3">
                {{ $post->id }}<br>
                {{ $post->title }}
            </div>
        @empty
            <div class="border rounded p-3">
                Empty
            </div>
        @endforelse
    </div>

    {{ $posts->links() }}
</div>
```

```php
<?php

namespace App\Livewire;

use Livewire\Component;
use Livewire\WithPagination;

class Posts extends Component
{
    use WithPagination;

    public function render()
    {
        return view('livewire.posts', [
            'posts' => \App\Models\Post::paginate(4),
        ]);
    }
}
```

**Comment by MomoDerDodo** [comment, marked as answer]
With the help of the Laravel Daily team the solution is found.

It's the Livewire version. Somehow when upgrading it turned out to be `livewire/livewire": "3.0"` flat instead it should be `livewire/livewire": "^3.0"` to include the latest fixes. After executing `composer update` all worked out as it should.

**Reply by agumaxwellforcode** [reply]
Did not work for me.
